src/cluster.c

```diff
@@ -2050,7 +2050,9 @@ int writeForgottenNodePingExt(clusterMsgPingExt **cursor, sds name, uint64_t ttl
     uint32_t extension_size = sizeof(clusterMsgPingExt) + sizeof(clusterMsgPingExtForgottenNode);
     (*cursor)->type = htons(CLUSTERMSG_EXT_TYPE_FORGOTTEN_NODE);
     (*cursor)->length = htonl(extension_size);
-    *cursor = (clusterMsgPingExt *) (ext->name + sizeof(clusterMsgPingExtForgottenNode));
+    /* Using ext+1 instead of ext->name+sizeof(clusterMsgPingExtForgottenNode)
+     * to avoid false positive sanitizer out-of-bounds error. */
```

Comment:
i'm not certain the comment is needed, the code now looks fairly simple.
maybe the comment is needed to explain the difference from writeHostnamePingExt.
maybe we can convert that one too:
```c
    *cursor = (clusterMsgPingExt *) ((intptr_t)ext + EIGHT_BYTE_ALIGN(sdslen(myself->hostname) + 1));
```